learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+
+The above three metrics are defined specifically with fairness in mind. By another view of fairness, though, `accuracy()` and `roc_auc()` are _also_ fairness metrics. Some stakeholders may believe that the most performant model regardless of group membership---i.e. the model that predicts readmission most accurately across groups---is the most fair model. 
+
+To evaluate each of these metrics against the specified workflows, we create a metric set like so:
+
+```{r}
+#| label: "m-set"
+m_set <-
+  metric_set(
+    accuracy,
+    roc_auc,
+    equal_opportunity(race),
+    equalized_odds(race),
+    demographic_parity(race)
+  )
+
+m_set
+```
+
+
+::: callout-note
+The first two inputs, `accuracy()` and `roc_auc()`, are standard [yardstick](https://yardstick.tidymodels.org) metrics. The latter three are also yardstick metrics like any other, though are created using the [_metric factories_](https://yardstick.tidymodels.org/dev/reference/new_groupwise_metric.html) `equal_opportunity()`, `equalized_odds()`, and `demographic_parity()`. When passed a data-column, metric factories output yardstick metrics. 
+:::
+
+<!-- TODO: remove the /dev from the above URL -->
+
+### Evaluation
+
+We can now evaluate the workflows we've defined against resamples using our metric set. The `workflow_map()` function will call `tune_grid()` on each workflow:
+
+```{r}
+#| eval: !expr eval_fits
+#| label: "wflow-map"
+set.seed(1)
+wflow_set_fit <- 
+  workflow_map(
+    wflow_set, 
+    verbose = TRUE, 
+    seed = 1, 
+    metrics = m_set,
+    resamples = readmission_folds
+  )
+```
+
+```{r}
+#| label: "save-wflow-set"
+#| eval: !expr eval_fits
+#| include: false
+save(wflow_set_fit, file = saved_fits)
+```
+
+A fitted workflow set looks just like the unfitted workflow set we saw previously, except that information on the tuning process is now stored in the `option` and `result` variables for each modeling workflow:
+
+```{r}
+#| label: "print-wflow-set"
+wflow_set_fit
+```
+
+## Model Selection
+
+Now that we've evaluated a number of models with a variety of metrics, we can explore the results to determine our optimal model. Beginning by a quick exploratory plot of our metrics:
+
+<!-- TODO: i'd really like to use `autoplot(wflow_set_fit, rank_metric = "roc_auc")` here, but the plot isn't usable without differentiating between the recipes. -->
+
+```{r}
+#| label: "plot-wflow-set-fit"
```

Comment:
I do think this plot could be clearer, like @topepo said earlier. The code does look really overwhelming, and the x-axis is kind of awkward. I do wonder about something more like a pairs plot for all the metrics.

Reply:
Actually, you're spot on. Let's just plot the metrics pairwise and then print out the most performant models from `roc_auc()` in a table. :) 474ea32!

Reply:
Okay, better. :) 91ca48a.